controllers/storageclaim_controller.go

```diff
@@ -595,8 +596,11 @@ func (r *StorageClaimReconciler) delete(obj client.Object) error {
 }
 
 func (r *StorageClaimReconciler) own(resource metav1.Object) error {
-	// Ensure StorageClaim ownership on a resource
-	return controllerutil.SetOwnerReference(r.storageClaim, resource, r.Scheme)
+	return r.ownWithOwner(r.storageClaim, resource)
+}
+
+func (r *StorageClaimReconciler) ownWithOwner(owner, dependent metav1.Object) error {
+	return controllerutil.SetControllerReference(owner, dependent, r.Scheme)
```

Comment:
This util does not make sense? what is the value added by it? why can we call `controllerutil.SetControllerReference(owner, dependent, r.Scheme)` on the call site? where we know the owner and the dependent.

In contrast, `r.own` has the added value of simplifying the operation of adding the object pointed by the reconciled event as the owner which is a common use case.

Reply:
Will be removed as asked in some other comment.

Reply:
done.